### PR TITLE
Release 0.2.1

### PR DESCRIPTION
**Pull request by oceans404**
## Add new Nada language features

- private equality
- left shift
- right shift
- probabilistic truncation

## Add `nada` SDK tool Improvements

- Include venv setup instructions
- Print program requirements
- Get a Nada program’s protocols model in text format
- Get a Nada program’s bytecode in text format

## Breaking changes
- Nillion Python Client user_id and party_id have changed from methods to attributes

## Sunsetting SecretArray in favor of Python lists
- Remove SecretArray references
- Add Python list examples and documentation

### CODE REVIEW

File: docs/nada-lang-tutorial-arithmetic-and-logic.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    inventory = Party(name="inventory")
+    accounting = Party(name="accounting")
+
+    price = PublicInteger(Input(name="price", party=pricing))
+    quantity = SecretInteger(Input(name="quantity", party=inventory))
+
+    revenue_in_cents = Integer(100) * price * quantity
+
+    return [Output(revenue_in_cents, "revenue_in_cents", accounting)]
+```-->
```
Comment (jimouris) [inline docs/nada-lang-tutorial-arithmetic-and-logic.md:68]: delete?

File: docs/nada-lang-tutorial-arithmetic-and-logic.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+## Boolean Values and Comparison of Integers
+
+Comparison operations can be applied to Nada integers. Such comparison expressions evaluate to Nada boolean values. Whether this resulting value is secret depends on whether the integers being compared are secret. Furthermore, Nada boolean values support the `if_else` method, which implements a variant of the [ternary conditional operator](https://en.wikipedia.org/wiki/Ternary_conditional_operator) that can work with Nada values (even if they are secret).
```
Comment (jimouris) [inline docs/nada-lang-tutorial-arithmetic-and-logic.md:72]: It's a bit confusing what is a "nada integer"; is it `Secret`, `Public`, or `literal`? Maybe we could just say "integers"?

File: docs/nada-lang-tutorial-arithmetic-and-logic.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    data_owner = Party(name="data_owner")
+
+    x = SecretInteger(Input(name="x", party=data_owner))
+    y = SecretInteger(Input(name="y", party=data_owner))
+
+    condition = x > y
+    maximum = condition.if_else(x, y)
+
+    return [Output(maximum, "maximum", data_owner)]
+```-->
```
Comment (jimouris) [inline docs/nada-lang-tutorial-arithmetic-and-logic.md:90]: delete?

File: docs/nada-lang-tutorial-arithmetic-and-logic.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    # Permitted.
+    a = x + Integer(123) + Integer(456)
+    b = x + Integer(123 + 456)
+
+    # Not permitted.
+    c = x + 123 + 456
+    d = x + 123
+
+    return [Output(a, "a", data_owner), Output(b, "b", data_owner)]
+```-->
```
Comment (jimouris) [inline docs/nada-lang-tutorial-arithmetic-and-logic.md:114]: delete?

File: docs/nada-lang-tutorial-lists-and-comprehensions.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# Lists and Comprehensions
+
+The Nada DSL supports the introduction and use of Python lists that contain secret integer values. This includes using a subset of the list comprehension syntax supported by Python.
+
+## Basic Example
+
+The program below uses a [Python list comprehensions](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) to build an ascending sequence of three secret integers in which the first entry is a secret integer input. This sequence of secret integers is then returned by the program.
```
Comment (jimouris) [inline docs/nada-lang-tutorial-lists-and-comprehensions.md:7]:
```suggestion
The program below uses [Python list comprehensions](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) to build an ascending sequence of three secret integers in which the first entry is a secret integer input. This sequence of secret integers is then returned by the program.
```

**Review by jimouris — APPROVED** [review]
LGTM! Should we remove the commented-out code? If something changes, the comments may be obsolete.

**Review by Davetbutler — APPROVED** [review]

File: docs/nada-lang-tutorial-arithmetic-and-logic.md (hunk 2 above)
Comment (oceans404) [reply docs/nada-lang-tutorial-arithmetic-and-logic.md:72]: updating to integers

**Pull request merged**